### PR TITLE
Fix: To attach alma hook if hook exist

### PR DESCRIPTION
**Pull request by Benjamin-Freoua-Alma**
### Reason for change

Fix issue with merchant where hooks aren't well hooked. Now we don't use the version of CMS but if the hook exist.

### Code changes

Add HookHelper to make the intelligents of the good hooks to hooked 

### How to test

_As a reviewer, you are encouraged to test the PR locally._

Install and/or reset the module to see if the goods are well hooked

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] You understand the impact of this PR on existing code/features

### Non applicable

- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: alma/alma.php [hunk 1]
```diff
@@ -41,12 +44,28 @@ class Alma extends PaymentModule
...
+     */
+    public $bootstrap;
+    /**
+     * @var int
+     */
+    private $is_eu_compatible;
+    /**
+     * @var string
+     */
+    public $confirmUninstall;
```
Comment (joyet-simon) [inline alma/alma.php:62]: Why have you got one variable in camel case and one in snake case?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v2.12.1
+
+- Fix to attach Alma hooks if hook exist
```
Comment (ClaireAlma) [inline CHANGELOG.md:5]: exists, and wich hook ? this description is not verbose :)

File: alma/alma.php [hunk 3]
```diff
@@ -41,12 +44,28 @@ class Alma extends PaymentModule
...
+     */
+    public $hook;
+    /**
+     * @var true
+     */
+    public $bootstrap;
+    /**
+     * @var int
+     */
+    private $is_eu_compatible;
```
Comment (ClaireAlma) [inline alma/alma.php:58]: It's better to add a description

File: alma/alma.php [hunk 4]
```diff
@@ -131,40 +152,14 @@ public function install()
     /**
      * Try to register mandatory hooks
      *
-     * @return bool as false if one registration fail
+     * @return bool
```
Comment (ClaireAlma) [inline alma/alma.php:155]: why remove the comment?

File: alma/lib/Model/HookHelper.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+     * @var array
+     */
+    private $hooks;
+
+    public function __construct()
+    {
+        $this->hooks = Hook::getHooks();
+    }
+
+    private function almaHooks()
```
Comment (ClaireAlma) [inline alma/lib/Model/HookHelper.php:48]: commentaire de fonction

File: alma/lib/Model/HookHelper.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        $existingHooks = $this->hooks;
+        $almaHooks = [
```
Comment (ClaireAlma) [inline alma/lib/Model/HookHelper.php:51]: pas très propre, mettre ça dans une variable de classe plutot

File: alma/lib/Model/HookHelper.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            'paymentOptions', // >= 1.7
+            'paymentReturn', // >= 1.7
+            'displayPayment',
+            'displayPaymentEU',
+            'displayPaymentReturn',
+            'displayProductPriceBlock', // >= 1.6
+            'displayProductButtons',
+        ];
+
+        foreach ($existingHooks as $hook) {
```
Comment (ClaireAlma) [inline alma/lib/Model/HookHelper.php:70]: soit mettres des continue, soit un switch case

File: alma/lib/Model/HookHelper.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+
+        return $almaHooks;
+    }
+
+    public function almaRegisterHooks()
+    {
+        $almaHooks = $this->almaHooks();
+        $existingHooks = $this->hooks;
```
Comment (ClaireAlma) [inline alma/lib/Model/HookHelper.php:88]: useless variable, use $this->hooks directly in the foreach

**Review by ClaireAlma — CHANGES_REQUESTED** [review]
We do the review in mob this afternoon?

**Review by ClaireAlma — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**